stuffer/s2n_stuffer.c

```diff
@@ -60,6 +60,12 @@ S2N_RESULT s2n_stuffer_reservation_validate(const struct s2n_stuffer_reservation
     return S2N_RESULT_OK;
 }
 
+/**
+ * Initialize a stuffer to reference some mutable data `in`.
```

Comment:
I know the function signature specifies that the data is mutable, but often we just assume it's not because there's no immutable version :grimacing: I wonder if you should remove the "mutable".

Same for blob down below.